### PR TITLE
Exponentiation in snark post

### PR DESCRIPTION
**Pull request by imeckler**
Screen Shot 2019-03-11 at 1 39 45 PM
Screen Shot 2019-03-11 at 1 39 30 PM

This PR adds a blog post about computing $(1/2)^x$ inside the snark and also adds an index page for the blog.

### CODE REVIEW

File: src/app/website/posts/taylor.markdown [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+It's important to use a threshold of this form because it means that the density
+of blocks over time does not depend on the distribution of stake.
+
+If you know anything about SNARKs, you know that inside of a SNARK all we
+can do is arithmetic (that is, addition and multiplication) in a finite field $F_p$. It's not at
+all clear how we can compute a fractional number raised to a fractional power!
+
+We'll go through a very cool technique for computing such a thing.
+All the code for doing what we'll talk about is implemented using
+[snarky](https://github.com/o1-labs/snarky) and can be found [here]().
```
Comment (bkase) [inline src/app/website/posts/taylor.markdown]: link missing at "here"

File: src/app/website/posts/taylor.markdown [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+The technique will go as follows:
+1. We'll use Taylor series to reduce the problem to doing arithmetic with
+  real numbers.
+2. We'll use approximate the arithmetic of real numbers using the arithmetic
```
Comment (bkase) [inline src/app/website/posts/taylor.markdown]: we'll use approximate -> we'll approximate

File: src/app/website/posts/taylor.markdown [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+arithmetic perfectly. However, there is a bit of an issue with this approach, which
+is that the integers involved get really huge really quickly when you add numbers together.
+For example,  $1/2 + 1/3 + ... + 1/n$ has $n!$ as its denominator, which is a large number.
+
+That's a problem for us inside the SNARK, because we're working with a finite field and
+want to make sure there is no overflow.
+
+A better approach is to use rational numbers whose denominator is a power of two.
+These numbers are called [dyadic rationals]() and are basically the same thing as
+[floating point numbers]().
```
Comment (bkase) [inline src/app/website/posts/taylor.markdown]: links

File: src/app/website/posts/taylor.markdown [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  again using continuity. Moreover, we saw that floating point arithmetic can easily
+  be simulated exactly using integer arithmetic.
+
+Now our final step is to simulate integer arithmetic using the arithmetic of
+our prime order field $\mathbb{F}_p$. But this step is actually the easiest!
+As long as we are careful about numbers not overflowing, it's the same thing.
+That is, if we know ahead of time that $a + b < p$, then we can safely
+compute $a + b \mod p$, knowing that the result will be the same as over the integers.
+The same is true for multiplication.
+
```
Comment (bkase) [inline src/app/website/posts/taylor.markdown]: Would be nice to add a conclusion I think

File: src/app/website/posts/taylor.markdown [hunk 5]
```diff
@@ -152,5 +165,10 @@ That is, if we know ahead of time that $a + b < p$, then we can safely
 compute $a + b \mod p$, knowing that the result will be the same as over the integers.
 The same is true for multiplication.
 
-[^0]: Actually it's a little more general than this, but for the sake of
-concreteness let's stick with this.
+# Conclusion
+
+Let's survey the net result of all this approximation: we have a very efficient way
```
Comment (bkase) [inline src/app/website/posts/taylor.markdown]: Capital `w` in we

Comment (imeckler) [reply src/app/website/posts/taylor.markdown]: done

**Pull request merged**